### PR TITLE
Frac_in_halos hotfix

### PR DESCRIPTION
**Pull request by SunilSimha**
`Mlow = Mhigh` shouldn't make the code barf.

### CODE REVIEW

File: frb/dm/igm.py [hunk 1]
```diff
@@ -342,8 +347,12 @@ def average_DMIGM(z, cosmo = defs.frb_cosmo, f_hot = 0.75, rmax=1., logMmin=10.3
     DM_IGM = DM_cosmic - DM_halos
 
     # Return
-    if cumul:
+    if cumul&return_DMhalos:
+        return DM_IGM, zeval, DM_halos
+    elif cumul&~return_DMhalos:
```
Comment (profxj) [inline frb/dm/igm.py:352]: I expect these are coded properly, but I get nervous seeing `&~`

can you put parenthesis around and use `not` instead?

File: frb/halos/hmf.py [hunk 2]
```diff
@@ -77,7 +77,13 @@ def frac_in_halos(zvals, Mlow, Mhigh, rmax=1.):
         ratios: ndarray
           rho_halo / rho_m
     """
-
+    # Deal with scalar input
+    if np.isscalar(zvals):
+        zvals = np.array([zvals])
```
Comment (profxj) [inline frb/halos/hmf.py:82]: do this:

`zvals = np.atleast_1d(zvals)`

File: frb/halos/hmf.py [hunk 3]
```diff
@@ -77,7 +77,13 @@ def frac_in_halos(zvals, Mlow, Mhigh, rmax=1.):
...
+    # Cheeky edge case
+    if Mlow == Mhigh:
```
Comment (profxj) [inline frb/halos/hmf.py:84]: are these floats?

if so, use  `np.isclose()`

**Review by profxj — CHANGES_REQUESTED** [review]
A few nits

File: frb/dm/igm.py (hunk 1 above)
Comment (SunilSimha) [reply frb/dm/igm.py:352]: Done.

File: frb/halos/hmf.py (hunk 3 above)
Comment (SunilSimha) [reply frb/halos/hmf.py:84]: Done.

File: frb/halos/hmf.py (hunk 2 above)
Comment (SunilSimha) [reply frb/halos/hmf.py:82]: Done.

**Review by profxj — APPROVED** [review]
all good

**Pull request merged**